Update local conda setup docs (#407)
AlexsLemonade/OpenScPCA-analysis: docs/technical-setup/environment-setup/setup-conda.md

- The [`conda-lock` package](https://conda.github.io/conda-lock/) is used to create fully reproducible, cross-platform [conda environments for analysis modules](../../contributing-to-analyses/determining-requirements/determining-software-requirements.md#managing-software-dependencies-with-conda)
- The [`jq` package](https://jqlang.github.io/jq/) provides JSON parsing capabilities
- The [`pre-commit` package](https://pre-commit.com) will allow you to use [pre-commit hooks when contributing to analysis modules](../../contributing-to-analyses/working-with-git/making-commits.md#pre-commit-checks)

<!-- Comment to force above to be bullets, next to be numbers -->

1. To create this environment, navigate in the terminal with `cd` to the `OpenScPCA-analysis` repository.
    - Again, if you open a [terminal in GitKraken](../../software-platforms/general-tools/using-the-terminal.md#gitkraken), you will automatically be in the repository folder.

1. Enter the following command in the terminal to create the `openscpca` environment and install the packages into it:

    ```sh
    conda env create -f environment.yml
    ```

>     When the process is complete, you'll see this message in the terminal providing the commands to activate and deactivate the environment:

Reviewer: ```suggestion
    You will see a number of status messages flow by as packages are downloaded and installed.
    When the process is complete, you'll see the following message in the terminal, which includes the commands to activate and deactivate the environment:
```

    ```{ .console .no-copy title="Output message after conda environment install"}
    # To activate this environment, use
    #
    #     $ conda activate openscpca
    #
    # To deactivate an active environment, use
    #
    #     $ conda deactivate
    ```

    ??? info "Prefer to modify your base environment instead?"
        While we recommended creating an `openscpca` environment, this is not strictly necessary.
        If you prefer to add these packages to your base environment, use this command instead:
        ```sh
        conda env update -n base -f environment.yml
        ```
        If you experience package conflicts when running this command, we recommend you go back to setting up a dedicated `openscpca` environment.

1. As indicated in the message, activate the environment with the following command:

    ```sh
    conda activate openscpca
    ```

>     At this point, your terminal prompt will have a prefix `(openscpca)`, which lets you know that you are in that conda environment.

Reviewer: ```suggestion
    At this point, your terminal prompt will show the prefix `(openscpca)`, which lets you know that you are in that conda environment.
```

    In general, this is how conda indicates which environment, if any, the terminal is working in.
    <figure markdown="span">
        ![Activate the openscpca environment](../../img/conda-activate-openscpca.png){width="300"}
    </figure>

    !!! tip "Activating the `openscpca` conda environment"

>         Whenever you are developing for OpenScPCA, you should work from this activated conda environment unless you are using a module-specific conda environment.

Reviewer: We should probably link to the docs about module-specific envs here
```suggestion
        Whenever you are developing for OpenScPCA, you should work from either the `openscpa` conda environment or a module-specific conda environment.
```

    Executing transaction: done
    # To activate this environment, use
    #
    #     $ conda activate openscpca
    #
    # To deactivate an active environment, use
    #
    #     $ conda deactivate
    ```

    ??? info "Prefer to modify your base environment instead?"
        While we recommended creating an `openscpca` environment, this is not strictly necessary.
        If you prefer to add these packages to your base environment, use this command instead:
        ```sh
        conda env update -n base -f environment.yml
        ```
        If you experience package conflicts when running this command, we recommend you go back to setting up a dedicated `openscpca` environment.

1. As indicated in the message, activate the environment with the following command:

    ```sh
    conda activate openscpca
    ```

    At this point, your terminal prompt will show the prefix `(openscpca)`, which lets you know that you are in that conda environment.
    In general, this is how conda indicates which environment, if any, the terminal is working in.
    <figure markdown="span">
        ![Activate the openscpca environment](../../img/conda-activate-openscpca.png){width="300"}

>     </figure>

Reviewer: Yeah, I have a feeling you don't need this. If you really wanted something, I might do the following chunk of code (basically reproducing your image) But I actually think the text you have is probably enough, especially given the potential variety in bash prompts.
````
```{ .console .no-copy title="Activated conda environment"}
bash$ conda activate openscpca
(openscpca) bash$ █
```
````

Author: > But I actually think the text you have is probably enough, especially given the potential variety in bash prompts.
